### PR TITLE
Date utility

### PR DESCRIPTION
**Pull request by MCurran16**
in the format function, let me know if I should do 'throw new Error' instead of a console.log if the parameter does not equal one of the accepted format types.

### CODE REVIEW

File: assets/js/lib/utils/date.js.coffee [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#= require lib/base.js.coffee
+#= require vendor/jquery.js
+
+do ->
+   FormattedDate = {}
+   today = new Date()
+
+   FormattedDate.weekDay = ->
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: I would make all these that return today's something specify today in their name so someone doesn't try to use them to convert a epoch time or something.

File: assets/js/lib/utils/date.js.coffee [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      return months[today.getMonth()]
+
+   FormattedDate.day = ->
+      return today.getDate()
+
+   FormattedDate.year = ->
+      return today.getFullYear()
+
+   # Formats: Short, Long, TimeStamp
+   FormattedDate.format = (format) ->
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: So did you copy someone else's code here? Why are you using 'self' as an actual variable?

File: assets/js/lib/utils/date.js.coffee [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#= require lib/base.js.coffee
+#= require vendor/jquery.js
+
+do ->
+   FormattedDate = {}
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: Don't think this is accurate. Lets call this DateUtils.

File: assets/js/lib/utils/date.js.coffee [hunk 4]
```diff
@@ -0,0 +1,42 @@
+#= require lib/base.js.coffee
+#= require vendor/jquery.js
+
+do ->
+   DateUtils = {}
+
+   DateUtils.weekDay = (date) ->
+      weekDays = ["Monday", "Tuesday", "Wednesday", \
+         "Thursday", "Friday", "Saturday", "Sunday"]
+      if date?
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: I would ditch this check and right at the top of the function just use assertArgs to do:
`assertArgs(arguments, Date)` to ensure a date format is coming in.

File: assets/js/lib/utils/date.js.coffee [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+         #{@day(@today)}, #{@year(@today)}"
+
+      weekday = @weekDay(dateVariable).substring(0,3)
+      month = @month(dateVariable).substring(0,3)
+
+      if date?
+         return "#{weekday}, #{month} #{@day(dateVariable)}, #{@year(dateVariable)}"
+      else
+         return defaultDate
+
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: Add extra newline here

File: assets/js/lib/utils/date.js.coffee [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+   define("lib.utils.DateUtils", DateUtils)
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: Add newline after this. (sometimes github cuts those off so ignore if it exist)

File: assets/js/lib/utils/date.js.coffee [hunk 7]
```diff
@@ -0,0 +1,42 @@
+#= require lib/base.js.coffee
+#= require vendor/jquery.js
+
+do ->
+   DateUtils = {}
+
+   DateUtils.weekDay = (date) ->
```
Comment (hunterbrowning) [inline assets/js/lib/utils/date.js.coffee]: I personally would prepend a "get" to all of these function names except 'format' so it becomes getWeekDay, etc. Just makes the code very readable in the implementation but its by no means an issue and stylistically your call.

File: assets/js/lib/utils/date.js.coffee (hunk 4 above)
Comment (MCurran16) [inline assets/js/lib/utils/date.js.coffee]: fixed

File: assets/js/lib/utils/date.js.coffee (hunk 5 above)
Comment (MCurran16) [inline assets/js/lib/utils/date.js.coffee]: fixed

File: assets/js/lib/utils/date.js.coffee (hunk 6 above)
Comment (MCurran16) [inline assets/js/lib/utils/date.js.coffee]: Ignored

**Comment by hunterbrowning** [comment]
LGTM!

**Pull request merged**